Workiva/platform_detect
# UIP-2192 Add browser "feature" detection, and optional CSS class "decorator" functionality

### ULTIMATE PROBLEM:
1. There was no way to detect whether a browser supports a specific "feature".
2. There was no easy way to append CSS classes on the root document node based on feature support, os, browser, browser version, etc.
    1. H5 libs like _Modernizr_ support this - so I feel as an opt-in feature, it adds value to this library as well.

### HOW IT WAS FIXED:
1. Add browser "feature" detection.
    1. By default, touch support is the only "feature" the lib will detect.
    2. See README changes in the diff for proposed API allowing consumers to add their own feature detection bits.
2. Added the `decorator.dart` library - which has a top-level function that consumers can invoke within `main()` that will "decorate" the `<html>` node with CSS classes that represent the meta that represents the os, browser, browser version, etc...

### TESTING INSTRUCTIONS:
1. Verify that unit tests are passing.
2. Pull down this branch, and run `pub get && pub serve example`
3. Navigate to http://localhost:8080
    1. Verify that the "HTML Decorators" portion of the page is emitting the correct CSS classes in a few different browsers
    2. Verify that when you enable "device mode" in chrome, and refresh the page, that the `touch` CSS class is present

---

__FYA:__ @greglittlefield-wf @travissanderson-wf @georgelesica-wf @jacehensley-wf @clairesarsam-wf 

__FYI:__ @joelleibow-wf @trentgrover-wf @evanweible-wf @maxwellpeterson-wf
